csharp/src/Google.Protobuf/Collections/MapField.cs

```diff
@@ -452,7 +452,17 @@ public void WriteTo(CodedOutputStream output, Codec codec)
             WriteContext.Initialize(output, out WriteContext ctx);
             try
             {
-                WriteTo(ref ctx, codec);
+                if (output.IsSerializationDeterministic())
+                {
+                    var sorted = new List<KeyValuePair<TKey, TValue>>(list);
+                    sorted.Sort((pair1, pair2) => Comparer<TKey>.Default.Compare(pair1.Key, pair2.Key));
```

Comment:
Using Comparer.Default begs the question as to whether the default string comparison is ordinal or not. I *believe* it is, but I'm not entirely sure.

I'd at least add a comment here - and possibly also note that map keys are guaranteed to be bools, integers or strings; never bytestrings or messages.

Reply:
It is NOT ordinal by default,  explicitly using StringComparer.Ordinal now